internal/util/observability/funccall.go

```diff
@@ -31,17 +31,19 @@ type funcCall struct {
 // FuncCall adds observability to a function call.
 //
 // It should be called at the very beginning of the function,
-// and returned function should be called at exit.
+// and returned [context.CancelFunc] function should be called at exit.
 // The returned function must not be passed or stored.
 // The only valid way to use FuncCall is:
 //
 //	func foo(ctx context.Context) {
-//	    defer FuncCall(ctx)()
-//	    // ...
+//		ctx, cancel := FuncCall(ctx)
+//		defer cancel()
+//		// ...
+//	}
 //
 // For the Go execution tracer, FuncCall creates a new region for the function call
 // and attaches it to the task in the context (or background task).
-func FuncCall(ctx context.Context) func() {
+func FuncCall(ctx context.Context) (context.Context, context.CancelFunc) {
```

Comment:
The documentation should be updated to mention:
* what context is returned;
* how it should be used;
* is it being canceled when returned function is called.

Reply:
@AlekSi could you please describe these things? E.g. in the task description or right here. Based on your comments, I see I don't understand how this function is expected to work and what idea is described in the task. Also I don't see what should be used instead of `go fc.run(stopCtx)`.

Reply:
> I see I don't understand how this function is expected to work

But you wrote that function?.. I don't follow. All I'm saying is that the function signature changed without a proper documentation update. Things that I wrote should described in the doc comment.

> Also I don't see what should be used instead of go fc.run(stopCtx).

I don't know. All I'm saying is that https://pkg.go.dev/runtime/trace#StartRegion description explicitly disallows the current implementation.